gcloud-java-resourcemanager/src/main/java/com/google/gcloud/resourcemanager/testing/LocalResourceManagerHelper.java

```diff
@@ -380,19 +390,30 @@ Response list(Map<String, Object> options) {
       if (includeProject) {
         count++;
         try {
-          projectsSerialized.add(jsonFactory.toString(extractFields(p, fields)));
+          projectsSerialized.add(jsonFactory.toString(extractFields(p, projectFields)));
         } catch (IOException e) {
           return Error.INTERNAL_ERROR.response(
               "Error when serializing project " + p.getProjectId());
         }
       }
     }
+    String[] listFields = (String[]) options.get("listFields");
     StringBuilder responseBody = new StringBuilder();
-    responseBody.append("{\"projects\": [");
-    Joiner.on(",").appendTo(responseBody, projectsSerialized);
-    responseBody.append(']');
-    if (nextPageToken != null) {
-      responseBody.append(", \"nextPageToken\": \"");
+    responseBody.append('{');
+    boolean commaNeeded = false;
+    // If fields parameter is set but no project field is selected we must return no projects.
+    if (!(projectFields != null && projectFields.length == 0)) {
+      responseBody.append("\"projects\": [");
+      Joiner.on(",").appendTo(responseBody, projectsSerialized);
+      responseBody.append(']');
+      commaNeeded = true;
+    }
+    if (nextPageToken != null && (listFields == null
+        || ImmutableSet.copyOf(listFields).contains("nextPageToken"))) {
+      if (commaNeeded) {
```

Comment:
Nit and fine as is but I would have used `responseBody.length() > 1` instead of the extra `commaNeeded` variable.